percent_encoding/src/lib.rs

```diff
@@ -176,40 +204,100 @@ static_assert! {
 /// Everything that is not an ASCII letter or digit.
 ///
 /// This is probably more eager than necessary in any context.
-pub const NON_ALPHANUMERIC: &AsciiSet = &CONTROLS
-    .add(b' ')
-    .add(b'!')
-    .add(b'"')
-    .add(b'#')
-    .add(b'$')
-    .add(b'%')
-    .add(b'&')
-    .add(b'\'')
-    .add(b'(')
-    .add(b')')
-    .add(b'*')
-    .add(b'+')
-    .add(b',')
-    .add(b'-')
-    .add(b'.')
-    .add(b'/')
-    .add(b':')
+pub const NON_ALPHANUMERIC: &AsciiSet = &ALPHA_NUM.complement();
+
+/// [RFC2396](https://datatracker.ietf.org/doc/html/rfc2396).
```

Comment:
non-blocking nit:

```suggestion
/// Lower case alpha numeric characters defined in [RFC2396](https://datatracker.ietf.org/doc/html/rfc2396#section-1.6).
```
A good description like this helps users searching for things (e.g. with a search engine), and the direct link to the section that these are defined in the RFC is useful when you need to dig into various things (obviously it's less useful for this particular case, but it's generally useful to have this as a general thing I think).